pkg/watermark/fetch/processor_to_fetch.go

```diff
@@ -98,6 +97,9 @@ func (p *ProcessorToFetch) IsDeleted() bool {
 }
 
 func (p *ProcessorToFetch) startTimeLineWatcher() {
+	if p.otWatcher == nil {
```

Comment:
Don't add the check, let it panic instead in case this really happens (even though it won't happen), otherwise it will help hide the issue.

Reply:
Makes sense. I am thinking I can keep the check but instead of `return`, explicitly panic. Please see https://github.com/numaproj/numaflow/pull/218#discussion_r996082512 on why I moved the null check from outside this method to inside.

Reply:
I decide to change it back. It seems we do have a case where the watcher can be nil and it's OK (heartbeat watcher for a source vertex).
```
        // we do not care about heartbeat watcher if this is a source vertex
	if v.hbWatcher != nil {
		go v.startHeatBeatWatcher()
	}
```
For such case, if I move null check from outside `startHeatBeatWatcher` to inside and panic, the thread will panic instead of skipping `startHeatBeatWatcher`. This could be a backward-incompatible change.

Reply:
that is a bug in our code, can you open an issue or PR for it? there should never be a nil watcher. We wrote no-op watcher and storer for that purpose.

Reply:
could you please create a new issue for this?

Reply:
New issue created: #230